### PR TITLE
add eigsh_lanczos to blocksparse backend

### PR DESCRIPTION
**Pull request by mganahl**

### CODE REVIEW

File: tensornetwork/backends/symmetric/symmetric_backend.py [hunk 1]
```diff
@@ -163,6 +163,141 @@ def conj(self, tensor: Tensor) -> Tensor:
...
+      A_vector_n.contiguous() #contiguous memory layout
+
+      # operate on tensor-data for scalar products
+      # this can be potentially problematic if vector_n and A_vector_n
+      # have non-matching shapes due to an erroneous matvec.
+      # If this is the case though an error will be thrown at line 281
+      diag_elements.append(
+          numpy.dot(numpy.conj(vector_n.data), A_vector_n.data))
+
+      if ((it > 0) and (it % ndiag) == 0) and (len(diag_elements) >= numeig):
```
Comment (Thenerdstation) [inline tensornetwork/backends/symmetric/symmetric_backend.py:269]: You can remove the parentheses.

File: tensornetwork/backends/symmetric/symmetric_backend.py [hunk 2]
```diff
@@ -163,6 +163,141 @@ def conj(self, tensor: Tensor) -> Tensor:
...
+        raise ValueError("if no `initial_state` is passed, then `shape` and"
+                         "`dtype` have to be provided")
+      initial_state = self.randn(shape, dtype)
+
+    if not isinstance(initial_state, BlockSparseTensor):
+      raise TypeError("Expected a `BlockSparseTensor`. Got {}".format(
+          type(initial_state)))
+
+    vector_n = initial_state
+    vector_n.contiguous()# bring into contiguous memory layout
```
Comment (Thenerdstation) [inline tensornetwork/backends/symmetric/symmetric_backend.py:234]: Nit: add space after and before # (there's several places).

File: tensornetwork/backends/symmetric/symmetric_backend.py [hunk 3]
```diff
@@ -163,6 +163,141 @@ def conj(self, tensor: Tensor) -> Tensor:
...
+        A_vector_n -= (krylov_vecs[-2] * norms_vector_n[-1])
+      else:
+        A_vector_n -= (krylov_vecs[-1] * diag_elements[-1])
+      vector_n = A_vector_n
+
+    A_tridiag = numpy.diag(diag_elements) + numpy.diag(
+        norms_vector_n[1:], 1) + numpy.diag(numpy.conj(norms_vector_n[1:]), -1)
+    eigvals, u = numpy.linalg.eigh(A_tridiag)
+    eigenvectors = []
+    if numpy.iscomplexobj(A_tridiag):
```
Comment (Thenerdstation) [inline tensornetwork/backends/symmetric/symmetric_backend.py:291]: Wouldn't it make sense to always do this, regardless of whether it's complex or not?

**Comment by Thenerdstation** [comment]
Update comments and I'll merge this in.

File: tensornetwork/backends/symmetric/symmetric_backend.py (hunk 3 above)
Comment (mganahl) [reply tensornetwork/backends/symmetric/symmetric_backend.py:291]: Yes, that's right. removed it

**Pull request merged**